backend/agent/interactem/agent/agent.py

```diff
@@ -163,10 +156,18 @@ def __init__(self, id: uuid.UUID, broker: NatsBroker):
         )
         self.agent_kv: KeyValueLoop[AgentVal]
         self.container_trackers: dict[RuntimeOperatorID, ContainerTracker] = {}
-        self._container_monitor_task: asyncio.Task | None = None
         # suppress monitor during container cleanup
         self._suppress_monitor = False
-        self._task_refs: set[asyncio.Task] = set()
+
+        # Deployment management
+        self._current_deployment: DeploymentManager | None = None
```

Comment:
The naming of this seem odd? Maybe `DeploymentManager` should names something else? Manager implies that is manages all deployments, but its associated with one what about `Deployment` of `DeploymentContext`?